src/main/java/com/gooddata/md/report/MetricElement.java

```diff
@@ -25,7 +25,7 @@
     private final String drillAcrossStepAttributeDisplayFormUri;
 
     @JsonCreator
-    public MetricElement(@JsonProperty("uri") final String uri,
+    MetricElement(@JsonProperty("uri") final String uri,
```

Comment:
we have never showed it, so i wouldn't call it "hide". more over i don't think we should let the user to handle bare string uris (thats why i added constructors with Metric class argument). and lastly constructor with four string arguments a bit cumbersome.